react-carousel-component/src/carousel.jsx

```diff
@@ -0,0 +1,83 @@
+import React from 'react';
+
+class Carousel extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { activeIndex: 0 };
+    this.revolver = this.revolver.bind(this);
+    this.nextImg = this.nextImg.bind(this);
+    this.prevImg = this.prevImg.bind(this);
+  }
+
+  revolver() {
+    this.intervalId = setInterval(this.nextImg, 3000);
+  }
+
+  nextImg() {
+    this.setState({ activeIndex: this.state.activeIndex + 1 });
+    if (this.state.activeIndex === this.props.images.length - 1) {
+      this.setState({ activeIndex: 0 });
+    }
+  }
+
+  prevImg() {
+    this.setState({ activeIndex: this.state.activeIndex - 1 });
+    if (this.state.activeIndex <= 0) {
+      this.setState({ activeIndex: this.props.images.length - 1 });
+    }
+  }
```

Comment:
Same here, figure out the next index then set the state.
```JAVASCRIPT
prevImg() {
  let activeIndex = this.state.activeIndex - 1;
  if (activeIndex < 0) {
    activeIndex = this.props.images.length - 1;
  }
  this.setState({ activeIndex: activeIndex });
}
```

Reply:
![Kapture 2022-06-14 at 14 55 03](https://user-images.githubusercontent.com/20695764/173696188-46662603-08dd-4525-97d5-744c84232191.gif)